mlflow/models/evaluation/base.py

```diff
@@ -107,7 +109,8 @@ def load(cls, path):
             uri = meta["uri"]
             ArtifactCls = _get_class_from_string(meta["class_name"])
             artifact = ArtifactCls(uri=uri)
-            artifact._load(os.path.join(artifacts_dir, artifact_name))
+            filename = posixpath.basename(urllib.parse.urlparse(uri).path)
```

Comment:
The OSS team agreed to gradually transition the use of path access to PathLib. It handles paths, especially Windows paths better. Does it work here? https://docs.python.org/3/library/pathlib.html

Reply:
Sounds good!